### ISSUE TITLE
LibreSSL support

### ISSUE THREAD

**Issue by loskiq** [libevent/libevent#743]
When compiling libevent with LibreSSL instead of OpenSSL, the following error occurs:

```
make[1]: Entering directory '/root/libevent-2.1.8-stable'
  CCLD     sample/le-proxy
./.libs/libevent_openssl.so: undefined reference to `BIO_get_init'
collect2: error: ld returned 1 exit status
Makefile:1439: recipe for target 'sample/le-proxy' failed
make[1]: *** [sample/le-proxy] Error 1
make[1]: Leaving directory '/root/libevent-2.1.8-stable'
Makefile:1248: recipe for target 'all' failed
make: *** [all] Error 2
```

[configure.txt](https://github.com/libevent/libevent/files/2759197/configure.txt)
[make.txt](https://github.com/libevent/libevent/files/2759198/make.txt)

**Comment by azat** [comment]
> When compiling libevent with LibreSSL instead of OpenSSL, the following error occurs:

Should be fixed in 28b807540

**Comment by azat** [comment]
@loskiq please reopen the issue if the referenced patch will not help you
